MdeModulePkg/Core/Dxe/Mem/Page.c

```diff
@@ -1765,6 +1772,157 @@ CoreFreePages (
   return Status;
 }
 
+/**
+  Frees previous allocated pages.
+
+  @param  Memory                 Base address of memory being freed
+  @param  NumberOfPages          The number of pages to free
+
+  @retval EFI_NOT_FOUND          Could not find the entry that covers the range
+  @retval EFI_INVALID_PARAMETER  Address not aligned or NumberOfPages is 0
+  @return EFI_SUCCESS            Pages successfully freed.
+
+**/
+EFI_STATUS
+EFIAPI
+CoreFreePages (
+  IN EFI_PHYSICAL_ADDRESS  Memory,
+  IN UINTN                 NumberOfPages
+  )
+{
+  EFI_STATUS                       Status;
+  EFI_MEMORY_TYPE                  MemoryType;
+  UINTN                            Alignment;
+  BOOLEAN                          IsGuarded;
+  UINTN                            NumberOfPagesAligned;
+  EFI_GCD_MEMORY_SPACE_DESCRIPTOR  Descriptor;
+  UINT64                           CurrentAddress;
+  UINT64                           EndAddress;
+  UINT64                           DescEnd;
+  UINT64                           CurrentLength;
+  EFI_STATUS                       OperationStatus;
+
+  NumberOfPagesAligned = NumberOfPages;
+  OperationStatus      = EFI_SUCCESS;
+
+  if (NumberOfPages == 0) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  CoreAcquireMemoryLock ();

```

Comment:
Should the acquire/release lock just be moved into CoreFindEntryToFree() and this API be free of locks.

Reply:
I’m fine to move the locks. The only reason I didn’t put them in CoreFindEntryToFree() was to allow future use of that API from locked functions, but since there is no usage today and it is easy to change, I’ll move it.

Reply:
I moved the locks to CoreFindEntryToFree.